cmd/tailscale/cli/up.go

```diff
@@ -221,9 +224,17 @@ func runUp(ctx context.Context, args []string) error {
 			if n.ErrMessage != nil {
 				fatalf("backend error: %v\n", *n.ErrMessage)
 			}
-			if s := n.State; s != nil {
+			if upArgs.forceReauth && !loggedin {
+				loggedin = true
```

Comment:
just getting a Notify message doesn't necessarily mean that you're logged in.

It could just be a status update.

You don't know whether you're logged in until you see the State/Status message.

Reply:
Sorry the variable name was misunderstanding. It should be `loginStarted` and it is used to make sure the login process happens exactly once.